entrypoint.sh

```diff
@@ -1238,17 +1238,52 @@ function prepare_gcc() {
 
     ALT_GCC_PRIO=200
     proc_version=$(cat /proc/version)
-    kernel_gcc_ver=$(echo ${proc_version} | grep -o 'gcc-[0-9]*')
+    # Example outputs:
+    # Ubuntu: Linux version 6.6.87.1-microsoft-standard-WSL2 (root@af282157c79e) (gcc (GCC) 11.2.0, GNU ld (GNU Binutils) 2.37) #1 SMP PREEMPT_DYNAMIC Mon Apr 21 17:08:54 UTC 2025
+    # SLES: Linux version 6.4.0-150600.21-default (geeko@buildhost) (gcc (SUSE Linux) 7.5.0, GNU ld (GNU Binutils; SUSE Linux Enterprise 15) 2.41.0.20230908-150100.7.46) #1 SMP PREEMPT_DYNAMIC Thu May 16 11:09:22 UTC 2024 (36c1e09)
+    # RHEL: Linux version 5.14.0-570.12.1.el9_6.x86_64 (mockbuild@x86-64-03.build.eng.rdu2.redhat.com) (gcc (GCC) 11.5.0 20240719 (Red Hat 11.5.0-5), GNU ld version 2.35.2-63.el9) #1 SMP PREEMPT_DYNAMIC Fri Apr 4 10:41:31 EDT 2025
+    
+    # Extract GCC version - flexible regex to catch anything between "gcc" and version number
+    gcc_version=$(echo "${proc_version}" | grep -oiE 'gcc[^0-9]*[0-9]+\.[0-9]+\.[0-9]+' | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
 
     debug_print "/proc/version: ${proc_version}"
 
-    if [ ! -z "${kernel_gcc_ver}" ]; then
-        debug_print "Kernel compiled with ${kernel_gcc_ver}"
+    if [ ! -z "${gcc_version}" ]; then
+        # Extract major version number (e.g., "11" from "11.2.0")
+        gcc_major_ver=$(echo "${gcc_version}" | cut -d. -f1)
+
+        debug_print "Kernel compiled with GCC version ${gcc_version} (major: ${gcc_major_ver})"
+
+        if ${IS_OS_UBUNTU}; then
+            kernel_gcc_ver="gcc-${gcc_major_ver}"
+            exec_cmd "apt-get -yq update; apt-get -yq install ${kernel_gcc_ver}"
+            gcc_binary="/usr/bin/${kernel_gcc_ver}"
+        elif ${IS_OS_SLES}; then
+            kernel_gcc_ver_package="gcc${gcc_major_ver}"
+            kernel_gcc_ver_bin="gcc-${gcc_major_ver}"
+            exec_cmd "zypper --non-interactive install --no-recommends ${kernel_gcc_ver_package}"
+            gcc_binary="/usr/bin/${kernel_gcc_ver_bin}"
+        else
```

Comment:
For OCP we don't need to install GCC. Compilation is done in a different container (DTK) which should have the right version installed.